lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -386,16 +390,10 @@ def __call__(self):
 
         choice_map.get(self.state)()
 
-    def __isSwarmManager(self):
-        try:
-            data = self.client.inspect_swarm()
-            json_str = json.dumps(data, ensure_ascii=False)
-            self.swarm_info = json.loads(json_str)
-            return True
-        except APIError:
-            return False
-
     def inspect_swarm(self):
+        self.client.module.deprecate(
```

Comment:
This is now called every time the swarm is updated. It should only be called if `state` is `inspect`.

Reply:
Adding a `if self.state == 'inspect': ` to `def __call__()` and putting the deprecation warning there (before the dispatching is done) is probably the cleanest solution.